File: book-service/src/main/java/com/book/bookservice/service/BookService.java
package com.book.bookservice.service;

import java.math.BigDecimal;
import java.util.List;

import com.book.bookservice.entity.Book;

public interface BookService {

	public List<Book> getBooks();

	public Book getBookById(Integer id);

	public Book addBook(Book book);

	public List<Book> searchBooksByCategoryAuthorPricePublisher(String category, String author, BigDecimal price,
			String publisher);

	public void delete(Integer userid);

}

File: src/app/service/book.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import Author from '../entity/Author';
import Book from '../entity/Book';
const GET_URL="http://localhost:8085/api/v1/digitalbooks/books/";
const POST_URL="http://localhost:8085/api/v1/digitalbooks/author/books/";
//const SEARCH_URL= "http://localhost:8085/api/v1/digitalbooks/books/search?category=NOVEL&author=John William&price=500&publisher=Publisher1";
@Injectable({
  providedIn: 'root'
})
export class BookService {

  constructor(public client :HttpClient) { }
  saveBook(book: Book){
    return this.client.post(POST_URL, book);
  
  }

  getBooks(){
    return this.client.get(GET_URL);
  }
  deleteBook(bookid: number) {
    return this.client.delete(GET_URL + bookid);
  }

  searchBook(category: string, author:Author, price:number, publisher:string){
    //return this.client.get(GET_URL+"search?category="+category+"&author="+author+"&price="+price+"&publisher="+publisher);
    return this.client.get(GET_URL+"search?category="+category+"&author="+author.name+"&price="+price+"&publisher="+publisher);
    
    //return this.client.get(SEARCH_URL);
  }

}
